src/lib.rs

```diff
@@ -243,6 +262,30 @@ impl AstarteSdk {
         Ok(())
     }
 
+    pub async fn remove_interface(&self, interface_name: &str) -> Result<(), AstarteError> {
+        // We Define a scope to isolate the lock and drop it before updating the introspection
+        self.remove_interface_from_map(interface_name).await?;
+        self.remove_properties_from_store(interface_name).await?;
+        self.send_introspection().await?;
+        Ok(())
+    }
+
+    async fn remove_interface_from_map(&self, interface_name: &str) -> Result<(), AstarteError> {
+        let interfaces_lock = self.interfaces.clone();
+        let mut interfaces = interfaces_lock.write().await;
+        let interfaces_map = &mut interfaces.interfaces;
+        let interface = interfaces_map
+            .get(interface_name)
+            .ok_or_else(|| AstarteError::SendError("Interface doesn't exist".into()))?;
```

Comment:
1. Use `to_string()` to convert str to String.
2. Maybe is better add a new error to `interface::Error` like `InterfaceNotFound`  and add it among the `AstarteError` enum.